src/main/webapp/app/exercises/programming/shared/code-editor/ace/code-editor-ace.component.ts

```diff
@@ -194,20 +194,22 @@ export class CodeEditorAceComponent implements AfterViewInit, OnChanges, OnDestr
      */
     onFileTextChanged(code: string) {
         /** Is the code different to what we have on our session? This prevents us from saving when a file is loaded **/
```

Comment:
Not added in this PR, but I would suggest to remove/refactor this comment as I have no clue what session this is about.

Reply:
Yeah, this comment is vague, however, we were not sure exactly how to define the behaviour in a better way here. The fileSession is the cache basically where we keep track of the file's contents so that we do not have to call the `getFileContent` from the server every time.